src/server/chromium/crNetworkManager.ts

```diff
@@ -463,11 +472,17 @@ class InterceptableRequest implements network.RouteDelegate {
     });
     if (!interceptPromise)
       return null;
-    const event = await interceptPromise;
-    return new network.InterceptedResponse(this.request, event.responseStatusCode!, event.responseErrorReason!, event.responseHeaders!);
+    const { event, redirectedRequest } = await interceptPromise;
+    if (this !== redirectedRequest)
+      this._redirectedRequest = redirectedRequest;
+    return new network.InterceptedResponse(redirectedRequest.request, event.responseStatusCode!, event.responseErrorReason!, event.responseHeaders!);
   }
 
   async fulfill(response: types.NormalizedFulfillResponse) {
+    if (this._redirectedRequest) {
```

Comment:
I don't understand this logic, unfortunately.

Reply:
If the request is redirected we always delegate the action to the last intercepted request in the chain as the client always holds on to the original request.

Reply:
Refactored the code as discussed in person.